cmd/paratime/denom/add.go

```diff
@@ -0,0 +1,49 @@
+package denom
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
+
+	cliConfig "github.com/oasisprotocol/cli/config"
+)
+
+var addDenomCmd = &cobra.Command{
+	Use:   "add <network> <paratime> <denomination> <number_of_decimals>",
+	Short: "Add a new denomination",
+	Args:  cobra.ExactArgs(4),
+	Run: func(cmd *cobra.Command, args []string) {
+		cfg := cliConfig.Global()
+		networkArg, ptArg, symbolArg, decimalsArg := args[0], args[1], args[2], args[3]
+
+		decimalsInt, err := strconv.Atoi(decimalsArg)
+		if err != nil {
+			cobra.CheckErr(fmt.Errorf("number of decimals '%s' can not be converted to integer", decimalsArg))
+			return
+		}
+
+		net := cfg.Networks.All[networkArg]
+		if net == nil {
+			cobra.CheckErr(fmt.Errorf("network '%s' does not exist", networkArg))
+			return
+		}
+
+		pt := net.ParaTimes.All[ptArg]
+		if pt == nil {
+			cobra.CheckErr(fmt.Errorf("pratime '%s' does not exist", ptArg))
+			return
+		}
+
+		denomInfo := &config.DenominationInfo{
+			Symbol:   symbolArg,
```

Comment:
Symbol and denomination identifier must be separate. Symbol could default to denomination identifier in case it is not specified.